src/Sparrow.Server/ByteString.cs

```diff
@@ -1379,40 +1375,54 @@ public void Release(ref ByteString value)
 
         private SegmentInformation AllocateSegment(int size)
         {
-            var memorySegment = Allocator.Allocate(size, AllocationFailed);
-            if (memorySegment.Segment == null)
-                ThrowInvalidMemorySegmentOnAllocation();
+            try
```

Comment:
Maybe it's worth to use a try-finally pattern like here:
https://github.com/ravendb/ravendb/pull/18565#discussion_r1611348786

Reply:
This needs to be looked with benchmark and assembler analysis in hand. Not a change to do here. try-finally while fast, may still be slower than what we have now.